src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -819,6 +824,80 @@ public BigDecimal getDecimalValue() throws IOException
         return _getBigDecimal();
     }
 
+    @Override
+    public LazyNumber getLazyNumber() throws IOException
+    {
+        LazyNumber lazyNumber = null;
+        if ((_numTypesValid & NR_BIGDECIMAL) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_BIGDECIMAL);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyBigDecimal(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_BIG_NUMBER_PARSER));
+                _numTypesValid |= NR_BIGDECIMAL;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else if ((_numTypesValid & NR_DOUBLE) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_DOUBLE);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyDouble(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_DOUBLE_PARSER));
+                _numTypesValid |= NR_DOUBLE;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else if ((_numTypesValid & NR_FLOAT) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_FLOAT);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyFloat(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_DOUBLE_PARSER));
+                _numTypesValid |= NR_FLOAT;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else if ((_numTypesValid & NR_BIGINT) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_BIGINT);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyBigInteger(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_BIG_NUMBER_PARSER));
+                _numTypesValid |= NR_BIGINT;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else {
+            throw new JsonParseException(this, "unable to convert value to LazyNumber: " + getText());
+        }
+        return lazyNumber;
+    }
+
+    private LazyNumber lazyNumberFromParsedPrimitiveNumber() throws IOException {
+        switch (_numTypesValid) {
+            case NR_INT:
+                return new LazyBigInteger(BigInteger.valueOf(_numberInt));
```

Comment:
Huh? Oh... I guess this is just because we do not have `LazyInteger` etc.

I guess this is possible approach but then requires `TokenBuffer` to check `getNumberType()` for `VALUE_NUMBER_INT` -- which may be sensible anyway.
But it should not then just blindly call `getLazyNumber()` to avoid unnecessary overhead going to `BigInteger` and back to `Integer` or `Long`.

Reply:
I've added LazyInteger and LazyLong now